src/xtgeo/grid3d/_egrid.py

```diff
@@ -691,19 +704,21 @@ def map_axis_units(self) -> Units:
         return self.egrid_head.mapunits
 
     @map_axis_units.setter
-    def map_axis_units(self, value):
+    def map_axis_units(self, value: Units) -> None:
         self.egrid_head.mapunits = value
 
     @property
     def grid_units(self) -> Units:
+        assert self.egrid_head.gridunit is not None
         return self.egrid_head.gridunit.unit
 
     @grid_units.setter
-    def grid_units(self, value):
+    def grid_units(self, value: Units) -> None:
+        assert self.egrid_head.gridunit is not None
```

Comment:
Again, unsure how to deal with these. Since the class has these as optional they can be ofc. be None. At runtime this would give an attribute error.

Reply:
I don't understand why the `assert...` is set?

Reply:
I guess we will discuss this tomorrow. Should probably hold off on merging until then.

Reply:
Lets wait until tomorrow, but the gist of it is that the [annotations](https://github.com/janbjorge/xtgeo/blob/cbc85054980534f02bcdb8ca7259ce5d29d53890/src/xtgeo/grid3d/_egrid.py#L527) indicates that the gridunit is optional. Since its optional we can not assign a value to it.